pkgs/development/python-modules/shiboken6/default.nix

```
@@ -47,6 +54,10 @@ stdenv'.mkDerivation rec {
     "-DBUILD_TESTS=OFF"
   ];
 
+  postPatch = ''
+    substituteInPlace cmake/ShibokenHelpers.cmake --replace '#!/bin/bash' "#!$SHELL"
```

```suggestion
  postConfigure = ''
    patchShebangs ../build/.qfp
```

is an alternative (haven't tested with cross compilation though), your option might be better

---

I'll compare, but patchShebangs is probably clearer so I'll probably switch to that (I vaguely recall thinking it expected to run scripts in the same phase they were generated, which is why I didn't do that to begin with, but I may have misremembered/misunderstood)

Also since this is only for build-time tooling and so both should choose the correct bash for cross

---

Yeah that seems to work fine and is probably clearer. I've changed it, thank you!

---

Ah, actually, it looks like downstreams (e.g. pyside6) would also need to do this every time. I'd rather patch this in `ShibokenHelpers.cmake`, but I'll change it so that it just gets the same bash from the environment too so there's no hardcoded shell paths in it

---

you can just inherit postConfigure in pyside6

---

I could, but anything using shiboken downstream could potentially have the same issue and inheriting the postConfigure would also mean something else shiboken-build-specific couldn't be added in it anymore

I'd really rather our shiboken package just work out of the box with something that's a potentially upstreamable change, rather than having to remember to patchShebangs downstreams

Would you be okay with what I currently have in `postPatch`?

Thank you again for the review!